HelpSource/Guides/WritingPrimitives.schelp

```diff
@@ -176,7 +176,18 @@ PyrObject *array2 = newPyrArray(g->gc, 2, 0, false);
 ...
 ::
 ::
-Care must be taken when writing utility functions which themselves create new objects, since this may happen somewhat opaquely and the calling context may not be known. Functions which may call themselves recursively also need special attention. In such cases setting teletype::runGC:: to false may be the safest option, or including a teletype::runGC:: arg so that GC behaviour is explicit. teletype::MsgToInt8Array:: is one example of such a function.
+One caveat: When making a new object reachable by storing it on the stack, you must ensure that you are not overwriting objects that will be needed later in the primitive. If this is done the receiver may be collected on any future allocations. One solution is to strong::push:: the object onto the stack, and then pop it when finished, e.g.:
+teletype::
+PyrSlot* rec = g->sp; // get the receiver
+PyrObject *result = newPyrArray(g->gc, 2, 0, true); // create the result
+++g->sp; SetObject(g->sp, result); // push the result array on the stack, so both it and rec are reachable
+... // further allocations which make use of the receiver to populate result
+--g->sp; // pop the stack back to the receiver slot since we stored result there above
+SetObject(rec, result); // now set the receiver
```

Comment:
since this supposed to be "model code":

- make the spacing of `*` in types consistent -- generally codebase seems to prefer 1 space to the left, 0 to the right, i.e. how `result` is declared
- avoid unnecessary abbreviation - `receiver` instead of `rec`
- place `SetObject(g->sp, result);` on a new line

Reply:
Done. Happy btw when you or anyone spots cosmetic things like this for you to just fix them in the online editor in cases where that would be faster than the discussion/agreement/commit/push cycle. There're enough big things to spend time on. :-)

Reply:
I can see all that. But I also worry about the balance between strict and consistent style and the ways in which this can be off-putting to casual contributors, especially if they're more relaxed about such things. As we really do want more people to be resolving issues I think we have to bear that in mind. Sometimes just doing things if they're trivial cosmetic fixes is a way of showing good faith on that and keeping people on board.

For myself, I'll never object to someone pushing something like that. I assume PRs are for (amongst other things) shared work on an issue. If it's sort of git manners thing that's stopping us from doing that, I'd say let's not worry about that. If it's big you can always post a warning and discuss, but these are not big.

Reply:
the bottleneck lately actually seems to be from a lack of reviews, not a lack of PR submissions.

Reply:
> the bottleneck lately actually seems to be from a lack of reviews, not a lack of PR submissions.

There remain 517 open issues, despite pruning. The issue this PR addresses was open for almost a year, and it's _really_ serious. The oldest issues are going on 6 years old.